compiler-plugin/src/main/kotlin/arrow/meta/phases/analysis/ElementScope.kt

```diff
@@ -276,12 +276,6 @@ interface ElementScope {
   
   fun String.blockCodeFragment(context: PsiElement?): Scope<KtBlockCodeFragment>
   
-  fun `if`(
-    condition: KtExpression,
-    thenExpr: KtExpression,
-    elseExpr: KtExpression? = null
-  ): IfExpression
```

Comment:
I'm starting to favor using Expression shapes over using some of these methods from `PsiFactory`. For example, there is an expectation that `condition` and `thenExpr` cannot be nullable for the following method:
```
    fun createIf(condition: KtExpression, thenExpr: KtExpression, elseExpr: KtExpression? = null): KtIfExpression {
        return (if (elseExpr != null)
            createExpressionByPattern("if ($0) $1 else $2", condition, thenExpr, elseExpr) as KtIfExpression
        else
            createExpressionByPattern("if ($0) $1", condition, thenExpr)) as KtIfExpression
    }
```
but this does not work well naturally with the PsiElement since the following are marked nullable:
```
public class KtIfExpression extends KtExpressionImpl {
    ...
    @Nullable @IfNotParsed
    public KtExpression getCondition() {
        return findExpressionUnder(KtNodeTypes.CONDITION);
    }
    ...
    @Nullable
    public KtExpression getThen() {
        return findExpressionUnder(KtNodeTypes.THEN);
    }
   ...
   @Nullable
    public KtExpression getElse() {
        return findExpressionUnder(KtNodeTypes.ELSE);
    }
}
```